### ISSUE TITLE
Change `Anvil` cross validation to work on whole dataset rather than splits

### ISSUE THREAD

**Issue by hmacdope** [OpenADMET/openadmet_models#357]
When we do our cross validation for both sklearn and pytorch  if you look closely at the implementations ([here](https://github.com/OpenADMET/openadmet-models/blob/b261cf805db20f2d186bf37f81915b0009f62ce7/openadmet/models/eval/cross_validation.py#L200-L202) and [here](https://github.com/OpenADMET/openadmet-models/blob/b261cf805db20f2d186bf37f81915b0009f62ce7/openadmet/models/eval/cross_validation.py#L538-L539) respectively) we are only doing our cross validation within the training set. 

That is to say the test set is left out of the cross validation altogether.

My understanding is that this is not necessarily wrong but possibly unusual. It should be a relatively simple change.
